Issue/2848 product variations scroll

ParaskP7 opened the pull request:
Fixes #2848
Prerequisite [WordPress-FluxC-Android #1786](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1786)

This change makes sure that the adapter is not set every time a new set of product variations are fetched from the database. The current situation was actually created by design since the ordering of product variations on the api's side and on the database' side was different. As such pagination could not work.

Both, this PR and [FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1786) together, solve this problem. The FluxC change aligns the ordering of product variations on both, the api and database, by 'date'. This, along with this PR's change, makes it possible for pagination to work as expected.

⚠️ Please note that the ordering of product variations now changes, from 'menu' to 'date', which change the product's logic for that screen. However, this change was communicated already with the team and everyone agreed on it.

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

ParaskP7 committed:
Fix product variations list scrolling to top when load more is triggered

This change makes sure that the adapter is not set every time a new set
of product variations are fetched from the database. The current
situation was actually created by design since the ordering of product
variations on the api's side and on the database' side was different. As
such pagination could not work.

This commit fixes partially this problem. Another change was already
made on the 'FluxC' side to align the ordering of product variations on
both, the api and database, by 'date'. Both changes together will make
it possible for pagination to work as expected. But, note that the
ordering of product variations now changes, from 'menu' to 'date', which
change the product's logic for that screen.

anitaa1990 reviewed (commented):
Hey @ParaskP7, the code changes look good! We can merge the PR once the FluxC hash has been updated 👍

ParaskP7 committed:
Merge remote-tracking branch 'origin/develop' into issue/2848-product-variations-scroll

ParaskP7 committed:
Updated fluxc commit hash to pull in product variations scroll fix

Merge commit: 'Merge pull request #1786 from wordpress-mobile/
fix/product-variations-scroll'

ParaskP7 commented:
> Hey @ParaskP7, the code changes look good! We can merge the PR once the FluxC hash has been updated 👍

Howdy @anitaa1990 and thanks for the review! 🤗 

I have just pushed a new commit which updates the `FluxC` version to the product variations scroll fix. Please let me know if I can further help progress this PR. 🙏

anitaa1990 committed:
Merge branch 'develop' into issue/2848-product-variations-scroll

anitaa1990 reviewed (approved):
LGTM :shipit:
